StreamSegmentContainerTests fails with unexpected exception

**Problem description**
```
io.pravega.segmentstore.server.containers.StreamSegmentContainerTests > testWriteFenceOut FAILED
    java.lang.AssertionError: Original container did not reject an append operation after being fenced out. Exception thrown was of unexpected type: io.pravega.common.ObjectClosedException: Object 'BlockingDrainingQueue' has been closed and cannot be accessed anymore.
        at org.junit.Assert.fail(Assert.java:88)
        at io.pravega.test.common.AssertExtensions.assertSuppliedFutureThrows(AssertExtensions.java:118)
        at io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.testWriteFenceOut(StreamSegmentContainerTests.java:1444)
```

**Problem location**
`StreamSegmentContainerTests.testWriteFenceOut`

**Suggestions for an improvement**
Add `ObjectClosedException` object closed exception to list of acceptable exceptions for the method to throw.

--- a/segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentContainerTests.java
+++ b/segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentContainerTests.java
@@ -13,6 +13,7 @@ import com.google.common.util.concurrent.AbstractService;
 import com.google.common.util.concurrent.Runnables;
 import com.google.common.util.concurrent.Service;
 import io.pravega.common.Exceptions;
+import io.pravega.common.ObjectClosedException;
 import io.pravega.common.TimeoutTimer;
 import io.pravega.common.concurrent.Futures;
 import io.pravega.common.io.StreamHelpers;
@@ -1444,8 +1445,9 @@ public class StreamSegmentContainerTests extends ThreadPooledTestSuite {
         AssertExtensions.assertSuppliedFutureThrows(
                 "Original container did not reject an append operation after being fenced out.",
                 () -> container1.append(segmentNames.get(0), new byte[1], null, TIMEOUT),
-                ex -> ex instanceof DataLogWriterNotPrimaryException
-                        || ex instanceof IllegalContainerStateException);
+                ex -> ex instanceof DataLogWriterNotPrimaryException      // Write fenced.
+                        || ex instanceof ObjectClosedException            // Write accepted, but OperationProcessor shuts down while processing it.
+                        || ex instanceof IllegalContainerStateException); // Write rejected due to Container not running.
 
         // Verify we can still write to the second container.
         container2.append(segmentNames.get(0), 0, new byte[1], null, TIMEOUT).get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);

